kernel/kernel-api/src/main/java/io/delta/kernel/Snapshot.java

```diff
@@ -29,13 +29,29 @@
 public interface Snapshot {
 
   /**
-   * Get the version of this snapshot in the table.
+   * Get the version of this snapshot in the table. Returns -1 if the table is empty and so has no
+   * commits.
    *
    * @param engine {@link Engine} instance to use in Delta Kernel.
    * @return version of this snapshot in the Delta table
    */
   long getVersion(Engine engine);
 
+  /**
+   * Get the timestamp (in milliseconds since the Unix epoch) of the latest commit of this snapshot.
+   * Returns -1 if the table is empty and so has no commits.
+   *
+   * <p>When InCommitTimestampTableFeature is enabled, the timestamp is retrieved from the
+   * CommitInfo of the latest commit which can result in an IO operation.
+   *
+   * <p>For non-ICT tables, this is the same as the file modification time of the latest commit in
+   * the snapshot.
```

Comment:
I actually wonder if we need this much detail here in a top-level API since this seems protocol related.  Maybe we can just say "returns the timestamp of this snapshot/version. This is determined according to the delta protocol ---link etc---"; and move these docs to the internal implementation for developers.